plugins/quickinspector/quickitemmodel.cpp

```diff
@@ -156,7 +155,14 @@ void QuickItemModel::connectItem(QQuickItem *item)
 
 void QuickItemModel::disconnectItem(QQuickItem *item)
 {
-    disconnect(item, nullptr, this, nullptr);
+    disconnect(item, SIGNAL(parentChanged(QQuickItem*)), this, SLOT(itemReparented()));
+    disconnect(item, SIGNAL(visibleChanged()), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(focusChanged(bool)), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(activeFocusChanged(bool)), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(widthChanged()), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(heightChanged()), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(xChanged()), this, SLOT(itemUpdated()));
+    disconnect(item, SIGNAL(yChanged()), this, SLOT(itemUpdated()));
```

Comment:
Good catch, one event filter per item is totally unnecessary here, We should get rid of that. But that's an unrelated improvement, so let's do that in a separate change.